### PR TITLE
Bootstrap4 upgrade

### PR DESCRIPTION
**Pull request by lawhead**

### CODE REVIEW

**Comment by lawhead** [comment]
vizr_update

**Comment by kirenotneb** [comment]
Looks good. I'm glad to see that adding the support for both versions was pretty trivial. I was concerned that it meant duplicating a lot of the UI elements.

File: js/components/ChartSummary.vue [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
         <caption>
           {{ messages.totalsCaption }}
-          <button class="copy-link pull-right btn btn-link btn-sm" @click="copyTable($refs.totalsTable)">
-            <span
-              class="glyphicon glyphicon-copy"
-              aria-hidden="true"
-              :title="messages.copyTable">
-            </span>
-          </button>
+          <a href="#" class="copy-link pull-right float-right" data-description="reload" role="button" @click.prevent="copyTable($refs.totalsTable)">
```
Comment (wsams) [inline js/components/ChartSummary.vue]: Is this the right `data-description`? I'm not following reload on the copy table button.

Comment (lawhead) [reply js/components/ChartSummary.vue]: Good catch. I think this was either a copy-paste error or leftover from a previous iteration. We're not actually using the data attribute anymore so I'll just remove it.

File: lib/metadata.php [hunk 2]
```diff
@@ -45,6 +45,12 @@
 $metadata_object->recordIdField = $record_id_field;
 $metadata_object->dataDictionary = $data_dictionary;
 $metadata_object->events = $event_map;
+if (REDCap::versionCompare ( REDCAP_VERSION , '8.7.1', '<')) {
```
Comment (wsams) [inline lib/metadata.php]: This is good to know.

Comment (lawhead) [reply lib/metadata.php]: Turns out I didn't need to use this, but I thought it was a potentially useful discovery so I left it in. Could be removed though.

**Pull request merged**